sklearn/linear_model/stochastic_gradient.py

```diff
@@ -66,6 +67,29 @@ def __init__(self, loss, penalty='l2', alpha=0.0001, C=1.0,
         self.warm_start = warm_start
         self.average = average
 
+        if n_iter is not None:
+            warnings.warn("n_iter parameter is deprecated in 0.19 and will be"
+                          " removed in 0.21. Use max_iter and tol instead.",
+                          DeprecationWarning)
+            # Same behavior as before 0.19
+            self.max_iter = n_iter
+            self.tol = -np.inf
```

Comment:
I copied the behavior of MLP-classes: 
>When the loss or score is not improving by at least tol for two consecutive iterations, convergence is considered to be reached and training stops.

In this case, a negative `tol` can be used (and there is no `ValueError` for negative `tol`).
The `-np.inf` is only here not to change the behavior of default params, but I can use `None` instead if it's less confusing.